src/cpp/core/system/PosixSystem.cpp

```diff
@@ -1264,19 +1298,49 @@ core::Error pidof(const std::string& process, std::vector<PidType>* pPids)
    return Success();
 }
 
-Error processInfo(const std::string& process, std::vector<ProcessInfo>* pInfo)
+Error processInfo(const std::string& process, std::vector<ProcessInfo>* pInfo, ProcessFilter filter)
 {
-   std::vector<PidType> pids;
-   Error error = pidof(process, &pids);
+   // use ps to capture process info
+   // output format
+   // USER:PID:PPID:PGID:PROCNAME
+   // we use a colon as the separator as it is not a valid path character in OSX
+   std::string cmd = process.empty() ? "ps acxj | awk '{OFS=\":\"; print $1,$2,$3,$4,$10}'"
```

Comment:
I'm doing it this way is it is done almost exactly the same way in the pidOf function. It also makes it a consistent command regardless of if we need to search by process name or not (since in both cases we use awk).